### DISCUSSION TITLE
Problem with Pasting on VS Code

### DISCUSSION THREAD

**Discussion by kichukic** [JaKooLit/Hyprland-Dots#384] [Q&A]
Hi guys im new to hyperland , after i installed this dot files i wont able to paste things from browser to vscode , but i was able to paste texts from terminal and other applications , but not from any browsers  , i tried all browsers  but still its not working  , is there anything that has to be done inside vscode ?, did anybody knows it ,it will be much helpfull

**Comment by JaKooLit** [comment]
from my discord server

This was just a repost 
original from here
https://discord.com/channels/1151869464405606400/1242108439866507284


Posting this as I have experience and lately some people experienced it

Background issues:
1.) Some weird Huge Border at the Pop up file manager when saving from Electron Based web browser, or attaching something to discord.
2.) Cant paste clipboard contents to and from any of this apps


The fix:
Launch the app with --enable-features=UseOzonePlatform --ozone-platform=wayland

ie. discord --enable-features=UseOzonePlatform --ozone-platform=wayland


Flatpak electron based apps does not seem to be affected by it

For Nixos:
Simply add this to your configuration.nix
environment.sessionVariables.NIXOS_OZONE_WL = "1";

**Reply by kichukic** [reply]
i have tried this 
code --enable-features=UseOzonePlatform --ozone-platform=wayland
and got this warning : 
Warning: 'enable-features' is not in the list of known options, but still passed to Electron/Chromium.

but still im not able to directly paste it  ,  but i can paste it from clipboard using windows+alt+v

**Comment by kichukic** [comment]
hi @JaKooLit   , is there any way to increase clipboard capacity cause its only copy half of contents that icopy fromany sources

**Reply by JaKooLit** [reply]
> hi @JaKooLit , is there any way to increase clipboard capacity cause its only copy half of contents that icopy fromany sources

you should approach the project page since I have NOT set any restrictions / settings on wl-clipboard

https://github.com/bugaevc/wl-clipboard

**Comment by M4700F** [comment]
Wayland has some copy-paste issues in VS Code. By forcing VS Code to use `x11` which is an older window system for Unix-like operating system with the flag `--ozone-platform=x11` , we can bypass this copy-paste issues that Wayland may have.

Just edit code.desktop by running `sudo nano /usr/share/applications/code.desktop` and add this flag `--ozone-platform=x11` under the [Desktop Entry] in the `Exec=/usr/share/code/code %F` line. Final line should look like this `Exec=/usr/share/code/code --ozone-platform=x11 %F`.

You can also see another section called [Desktop Action new-empty-window] where there is another `Exec` line. There you can also add this flag `--ozone-platform=x11`.

After that save the file. I think this should solve the issue.

**Reply by Glerme** [reply]
Its working in 2025, thanks @M4700F

**Reply by Henryyr** [reply]
Hi is there a way to put that line not only on Code configuration? Because when i copy from brave browser to terminal, whatsapp electon and vscode. It crash and when i see the clipboard (super + alt + v) its doesnt show up

**Reply by M4700F** [reply]
From what you're describing, it seems like a system wide clipboard issue if I am not wrong. One possible solution could be launching each electron app(like brave, whatsapp etc.) with `--ozone-platform=x11` flag.
You can do this by editing their `.desktop` files in `/usr/share/applications` just like we did for vs-code.

**Reply by Henryyr** [reply]
thanks!

**Reply by M4700F** [reply]
Hey I faced some problem regarding copying from brave to vs code.  So I edited the `brave-browser.desktop` by `sudo nano /usr/share/applications/brave-browser.desktop` and added `--ozone-platform=wayland` to every `Exec` line. And after that the copy from brave browser issue is resolved for me.

**Comment by piyushyadav0191** [comment]
@M4700F thanks it worked

**Reply by M4700F** [reply]
You are welcome. @piyushyadav0191
